boutdata/processor_rearrange.py

```diff
@@ -56,6 +56,17 @@ def get_processor_layout(boutfile, has_t_dimension=True, mxg=2, myg=2):
 
     """
 
+    if mxg is None:
+        try:
+            mxg = boutfile["MXG"]
+        except KeyError:
+            mxg = 2
+    if myg is None:
+        try:
+            myg = boutfile["MYG"]
+        except KeyError:
+            myg = 2
```

Comment:
It's possible to combine the logic into one-liners that aren't too bad:

```suggestion
    mxg = mxg or boutfile.get("MXG", 2)
    myg = myg or boutfile.get("MYG", 2)
```

Reply:
Needs a `DataFile.get()` method though! See https://github.com/boutproject/boututils/pull/38...

Reply:
Ah, sorry, I thought it was already sufficiently dict-like!